src/java/org/apache/cassandra/journal/AsyncCallbacks.java

```diff
@@ -17,15 +17,16 @@
  */
 package org.apache.cassandra.journal;
 
+import org.apache.cassandra.service.jounral.RecordPointer;
+
 public interface AsyncCallbacks<K, V>
 {
     /**
      * Invoked once an entry has been written to the file, and indexes have been updated, but before it
      * has been flushed to disk. Invoked from the writer thread. Execution order of onWrite() callbacks
      * with regard to each other is undefined.
      */
-    void onWrite(long segment, int position, int size, K key, V value, Object writeContext);
-
+    void onWrite(RecordPointer pointer, int size, K key, V request, Object writeContext);
```

Comment:
In retrospect, maybe better to leave it at primitive segment + position arguments, so we don't allocate an extra object for implementations that don't need it.